extensions/security/runtime-spi/src/main/java/io/quarkus/security/spi/runtime/SecurityEventHelper.java

```diff
@@ -0,0 +1,29 @@
+package io.quarkus.security.spi.runtime;
+
+import jakarta.enterprise.inject.spi.BeanManager;
+
+public record SecurityEventHelper(boolean fireEventOnSuccess, boolean fireEventOnFailure) {
+
+    public static SecurityEventHelper of(SecurityEvent successEvent, SecurityEvent failureEvent, BeanManager beanManager,
+            boolean enabled) {
+        if (enabled) {
+            boolean fireAllSecurityEvents = !beanManager.resolveObserverMethods(SecurityEvent.NOOP).isEmpty();
+            if (fireAllSecurityEvents) {
+                return new SecurityEventHelper(true, true);
+            } else {
+                return new SecurityEventHelper(
+                        !beanManager.resolveObserverMethods(successEvent).isEmpty(),
+                        !beanManager.resolveObserverMethods(failureEvent).isEmpty());
+            }
+        } else {
+            return new SecurityEventHelper(false, false);
+        }
+    }
+
+    /**
+     * @return true if given event type is observed and security events are enabled
+     */
+    public static boolean eventObserved(SecurityEvent event, BeanManager beanManager, boolean enabled) {
```

Comment:
I wonder if it should return some enum, say, `ObserveType`, with values like `NONE, ALL, SYNC, ASYNC`, so that rather than doing 2 calls `event.fire(); event.fireAsync();`, we can do for example, only `event.fire();` if no `ASYNC` observers are available ? Does it sound reasonable ?

Reply:
@michalvavrik Sounds goods, can you consider something like `SecurityEventHelper.fire(Event<? extends SecurityEvent> event)` and have both `fire()` and `fireAsync()` there ? I don't mean to be pedantic, it is just looks like that having two calls in every place where firing is required might be optimized a bit

Reply:
Sure, will do.